packages/core/src/network/network.ts

```diff
@@ -0,0 +1,361 @@
+
+            // Prepare agent context with any relevant state information
+            let agentContext = context.message;
+
+            // If this isn't the first agent call, we might want to include previous results
+            if (this.current_state && this.current_state.callCount > 0) {
+              // Log that we're passing context from previous agents
+              this.logger.debug(`Passing context from previous agents to ${agent.name}`);
+            }
+
+            // Generate response from the agent
+            const startTime = Date.now();
+
+            const result = await agent.generate(agentContext, {
+              context: this.current_state?.lastResult
+                ? [{ role: 'assistant', content: this.current_state?.lastResult }]
+                : [],
+            });
+
+            const duration = Date.now() - startTime;
+
+            // Update network state with this agent's result
+            if (this.current_state) {
+              this.current_state.lastResult = result.text;
+              this.current_state.callCount += 1;
+
+              // Store the result in the state object as well with a timestamp
+              const timestamp = new Date().toISOString();
+              this.current_state.state.set(agentId, {
+                result: result.text,
+                timestamp: timestamp,
+                input: agentContext,
+              });
+              this.logger.debug(
+                `Updated network state [agent=${agentId}] [callCount=${this.current_state.callCount}] [timestamp=${timestamp}]`,
+              );
+            }
+
+            // Log a preview of the result
+            const resultPreview = result.text.length > 100 ? `${result.text.substring(0, 100)}...` : result.text;
+            this.logger.debug(
+              `Agent response [name=${agent.name}] [duration=${duration}ms] [preview=${resultPreview}]`,
+            );
+            return { message: result.text };
+          },
+        });
+        return memo;
+      },
+      {} as Record<string, AgentTool>,
+    );
+
+    // Create the router agent with the tools
+    return {
+      ...agentAsTools,
+      getNetworkState: createTool({
+        id: 'getNetworkState',
+        description: 'Get the current state of the agent network, including previous agent results and call history',
+        inputSchema: z.object({}),
+        outputSchema: z.object({
+          state: z.object({
+            callCount: z.number(),
+            lastResult: z.any(),
+            input: z.string(),
+            agentResults: z.record(z.string()),
+            agentHistory: z.array(
+              z.object({
+                agent: z.string(),
+                input: z.string(),
+                result: z.string(),
+                timestamp: z.string(),
+              }),
+            ),
+            stateData: z.record(z.unknown()),
+          }),
+        }),
```

Comment:
This looks like it could use a huge amount of tokens - isn't this covered by the agent tool responses in each agent tool call? Since the tool call args and results will be in context for the router agent